### ISSUE TITLE
2.12-ification

### ISSUE THREAD

**Issue by mosesn** [codingwell/scala-guice#55]
We should start publishing against 2.12.0-M4 if possible, since it looks like 2.12.0 final is around the corner.

**Comment by tsuckow** [comment]
I seem to remember being unable to find a compatible version of some of our dependencies. This may have changed now.

**Comment by tsuckow** [comment]
https://github.com/tsuckow/scala-guice/tree/feature/2.12

I'd push a snapshot but I need to recover my credentials from my old box (Motherboard failed).

Note it fails a pedantic test:

```
[info] - should give a useful error when bound on itself *** FAILED ***
[info]   "net.codingwell.scalaguice.InternalModule.bind(ScalaModule.scala:69)" did not contain "ScalaModuleSpec.scala" (ScalaModuleSpec.scala:110)
```

**Comment by mosesn** [comment]
@tsuckow thanks!  we're going to need this to get finatra on the 2.12 train.

**Comment by tsuckow** [comment]
I think I resolved the issues with 2.12. Going to try to mix that in with the Guice 4.1 pull request for the next version (Probably this weekend sometime unless I get heatstroke)

**Comment by tsuckow** [comment]
I deployed a 4.1.0-SNAPSHOT on 2.12.0-M5 If you could give it a spin and see if it works that would be great. If there is no negative feedback I'll publish it proper.

**Comment by mosesn** [comment]
We're not ready for it yet (we have other dependencies which aren't on 2.12.x yet), but I wonder if there's another way I can help test it out?

**Comment by tsuckow** [comment]
If you could test on a version you do use 2.10 or 2.11 that would still be awesome. I made a few minor tweaks for 2.12. It passes all the tests but that isn't conclusive.

Those tweaks were the change I made for error messages. I am not sure it actually worked correctly in 4.0.1 and before, If you made a binding have a runtime problem, it should include the line of the binding in your code. Things like duplicate bindings I think would work for the test. There may be some spots where we could extend our tests to verify more of the possible runtime errors.

**Comment by tsuckow** [comment]
Released for 2.12.0-M5
